extension/src/extension.ts

```diff
@@ -28,8 +31,24 @@ export function activate(context: vscode.ExtensionContext) {
     const componentProvider = new ComponentViewerProvider(context);
     const registration = vscode.workspace.registerTextDocumentContentProvider("xc-preview", componentProvider);
 
-    const terminal = vscode.window.createTerminal("xcomponent");
-    context.subscriptions.push(terminal);
+    const buildTerminal = "build";
+    const webSocketBridgeTerminal = "webSocketBridge";
+    const spyTerminal = "spy";
+    const terminalNames = [buildTerminal, webSocketBridgeTerminal, spyTerminal];
+    const terminals: { [key: string]: vscode.Terminal } = {};
```

Comment:
You are in a typescript / node environment. Just use a Map !
const terminals: Map<string, vscode.Terminal>